materialize-bigquery/binding.go

```diff
@@ -1,225 +1,307 @@
 package main
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"strings"
 
-	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/storage"
+	"github.com/estuary/flow/go/protocols/fdb/tuple"
 	pf "github.com/estuary/flow/go/protocols/flow"
-	sqlDriver "github.com/estuary/flow/go/protocols/materialize/sql"
 )
 
-const tempTableNamePrefix = "flow_temp"
+type Binding struct {
+	// CreateSQL is a template SQL query that is generated
+	// based on the values for this binding. The value can be compiled when the
+	// Binding is initialized as there's no dynamic value in the
+	// query as it's moving data that is stored in cloud storage.
+	LoadSQL string
+
+	// CreateSQL is a template SQL query that is generated
+	// based on the values for this binding. The value can be compiled when the
+	// Binding is initialized as there's no dynamic value in the
+	// query as it's moving data that is stored in cloud storage.
+	CreateSQL string
+
+	// The google cloud storage bucket that is configured
+	// to work with this binding. It is expected to be
+	// configured and working when being assigned here.
+	bucket *storage.BucketHandle
+
+	// ExternalTableAlias is a string generated so that it can be used to reference
+	// the external cloud storage data in a SQL Query. At the core of this, there is data
+	// stored in Google Cloud Storage in a JSON format, and the connector needs to access that data to
+	// merge it into BigQuery's dataset. By defining this table alias here, the Queries stored above can
+	// use this alias when compiling the SQL queries, and job will be configured so that the query runs
```

Comment:
I was a bit confused when reading this: the queries above are already compiled when this struct is being initialised, so this property is not really used to generate those queries, right?

Reply:
The externalTableAlias is used as a configuration map for bigquery so it can locates the table, it's a readonly property that doesn't change once it's created.